classifiers/nlp/ZeroShotTFClassifier/__init__.py

```diff
@@ -0,0 +1,236 @@
+                :func: `~transformers.PreTrainedModel.save_pretrained`,
+                e.g.: `./my_model_directory/`.
+        :param base_tokenizer_model: The name of the base model to use for
+            creating the tokenizer. If None, will be equal to
+            `pretrained_model_name_or_path`.
+        :param pooling_strategy: the strategy to merge the word embeddings
+            into the chunk embedding. Supported strategies include 'cls',
+            'mean', 'max', 'min'.
+        :param layer_index: index of the transformer layer that is used
+            to create encodings. Layer 0 corresponds to the embeddings layer
+        :param max_length: the max length to truncate the tokenized
+            sequences to.
+        """
+
+        super().__init__(*args, **kwargs)
+        self.labels = labels
+        self.pretrained_model_name_or_path = pretrained_model_name_or_path
+        self.base_tokenizer_model = (
+            base_tokenizer_model or pretrained_model_name_or_path
+        )
+        self.pooling_strategy = pooling_strategy
+        self.layer_index = layer_index
+        self.max_length = max_length
+
+        if self.pooling_strategy == "auto":
+            self.pooling_strategy = "cls"
+            self.logger.warning(
+                '"auto" pooling_strategy is deprecated, Defaulting to '
+                ' "cls" to maintain the old default behavior.'
+            )
+
+        if self.pooling_strategy not in ["cls", "mean", "max", "min"]:
+            self.logger.error(
+                f"pooling strategy not found: {self.pooling_strategy}."
+                ' The allowed pooling strategies are'
+                '"cls", "mean", "max", "min".'
+            )
+            raise NotImplementedError
+
+        if len(self.labels) < 2:
+            raise ValueError("The number of target labels must be at least 2.")
+
+        if len(self.labels) != \
+                len(set(self.labels)):
+            raise ValueError(
+                "There are duplicate value in the target_label argument."
+            )
+
+    @batching
+    @as_ndarray
+    def predict(self,
+                data: "np.ndarray",
+                *args,
+                **kwargs) -> "np.ndarray":
+        """
+         Perform zero shot classification on `data`, the predicted label
+         for each sample in X is returned.
+
+         The output is a zero/one one-hot label for L-class multi-class
+         classification of size (B, L) with `B` being `data.shape[0]`
+         and `L` being the number of potential classification labels.
+
+         :param data: the input textual data to be classified, a 1 d
+            array of string type in size `B`
+         :type data: np.ndarray
+         :return: zero/one one-hot predicted label of each sample
+            in size `(B, L)`
+         :rtype: np.ndarray
+        """
+
+        data_encoded = self._encode(data)
+        labels_encoded = self._encode(self.labels)
```

Comment:
since the `labels` are constant, no need to encode them every time, I think in `post_init` u can keep an array of `encodings` for all the labels as an optimization